### PR TITLE
heppy updates for ICHEP from multileptons

### PR DESCRIPTION
**Pull request by peruzzim**
- soft muon ID, ICHEP version
- WP encoding modified for tau ID MVAOldDMRun2, dR=0.3 and 0.5

### CODE REVIEW

File: PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py [hunk 1]
```diff
@@ -84,8 +87,8 @@ def id6(tau,X):
             tau.idMVA = id6(tau, "by%sIsolationMVArun2v1DBoldDMwLT")
             tau.idMVANewDM = id6(tau, "by%sIsolationMVArun2v1DBnewDMwLT")
             tau.idCI3hit = id3(tau, "by%sCombinedIsolationDeltaBetaCorr3Hits")
-            tau.idMVAOldDMRun2 = id3(tau, "by%sIsolationMVArun2v1DBoldDMwLT")
-            tau.idMVAOldDMRun2dR03 = id3(tau, "by%sIsolationMVArun2v1DBdR03oldDMwLT")
+            tau.idMVAOldDMRun2 = id5(tau, "by%sIsolationMVArun2v1DBoldDMwLT")
```
Comment (peruzzim) [inline PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py]: Warning: the encoding of tau.idMVAOldDMRun2 is not backward compatible (i.e. the cut value should be changed downstream to keep the same selection)

**Comment by gpetruc** [comment]
@steggema: objections?

File: PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py (hunk 1 above)
Comment (steggema) [inline PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py]: Not sure why this line was introduced - it's the same as line 87 but only saved 3 working points; now saves 5 working points. I would remove it

File: PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py [hunk 2]
```diff
@@ -84,8 +87,8 @@ def id6(tau,X):
...
+            tau.idMVAOldDMRun2dR03 = id4(tau, "by%sIsolationMVArun2v1DBdR03oldDMwLT")
```
Comment (steggema) [inline PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py]: The deltaR 03 MVA isolation should also have 6 working points saved, so no need to introduce id4

File: PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py (hunk 1 above)
Comment (peruzzim) [reply PhysicsTools/Heppy/python/analyzers/objects/TauAnalyzer.py]: Historical reasons, now it should be fixed (just new dR=0.3 ID added with 6 working points).

@gpetruc, please wait for merging, will require to adapt TauType in cmgtools.

**Pull request merged**